src/Database/Mysql/Driver/Mysql.php

```diff
@@ -54,15 +59,46 @@ public function update(Tenant $tenant): bool
             return false;
         }
 
-        return $this->process($tenant, [
-            'user' => "RENAME USER `{$config['oldUsername']}`@'{$config['host']}' TO `{$config['username']}`@'{$config['host']}'",
-        ]);
+        if (isset($config['allowedHost'])) {
+            $config['host'] = $config['allowedHost'];
+        }
+
+        $tempTenant = $tenant;
+        $tempTenant->{$tempTenant->getTenantKeyName()} = $tenant->getOriginal($tenant->getTenantKeyName());
+        $originalTenant = Tenancy::getTenant();
+
+        Tenancy::setTenant($tempTenant);
+        $connection = Tenancy::getTenantConnection();
+        $tables = $connection->select('SHOW TABLES');
+        Tenancy::setTenant($originalTenant);
+
+        $tableStatements = [];
+        foreach ($tables as $table) {
+            foreach ($table as $key => $value) {
+                $tableStatements['table'.$value] = "RENAME TABLE `{$config['oldUsername']}`.{$value} TO `{$config['database']}`.{$value}";
+            }
+        }
+        // Add database drop statement as last statement
+        $tableStatements['delete_db'] = "DROP DATABASE `{$config['oldUsername']}`";
```

Comment:
Why not move this to below the array_merge for the `$statements` array?

Reply:
Do you mean to make another array_merge under the current one?
If you mean why I didn't do it like this:
```
        $statements = array_merge([
            'user'     => "RENAME USER `{$config['oldUsername']}`@'{$config['host']}' TO `{$config['username']}`@'{$config['host']}'",
            'password' => "ALTER USER `{$config['username']}`@`{$config['host']}` IDENTIFIED BY '{$config['password']}'",
            'database' => "CREATE DATABASE `{$config['database']}`",
            'grant'    => "GRANT ALL ON `{$config['database']}`.* TO `{$config['username']}`@'{$config['host']}'",
            'delete_db' => "DROP DATABASE `{$config['oldUsername']}`",
        ], $tableStatements);

Is because I'm afraid this will drop the database before moving the tables.